scipy/signal/tests/test_signaltools.py

```diff
@@ -267,6 +267,19 @@ def test_fillvalue(self):
                    [32, 46, 67, 62, 48]])
         assert_array_equal(c, d)
 
+    def test_fillvalue_direct_cast(self):
+        # These is an example that used to work, but it seems odd enough
+        # to disallow directly?
```

Comment:
wait, I can get the old behaviour with correct warning if I change the code slightly, that is probably better.